storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransaction.java

```diff
@@ -425,4 +431,10 @@ public Optional<SignedBeaconBlock> getBlockIfAvailable(final Bytes32 blockRoot)
         .map(SignedBlockAndState::getBlock)
         .or(() -> store.getBlockIfAvailable(blockRoot));
   }
+
+  @Override
+  public SafeFuture<Optional<List<BlobSidecar>>> retrieveBlobSidecars(
```

Comment:
not sure here.
generally there is the fallback to store logic in case the tx doesn't contain it. The idea is that a `StoreTranscation` is also a `Store` but with some temporary data overlaying it.

Reply:
Still don't understand this

Reply:
```java
    final Optional<List<BlobSidecar>> maybeBlobSidecars =
        Optional.ofNullable(blobSidecars.get(slotAndBlockRoot));
    if (maybeBlobSidecars.isPresent()) {
      return SafeFuture.completedFuture(maybeBlobSidecars);
    }
    return store.retrieveBlobSidecars(slotAndBlockRoot);
```